### PR TITLE
feat(experiments): Bring funnel options to the experiment

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

This PR brings a few enhancements to experiments:

1. Moves participant selection and flag targeting type to the feature flag. As we add new features, keeping these in sync is a losing battle, so from now on, all changes for the flag will happen on the flag itself. This also helps cleanup lots of code in experiments.
image

2. Add funnel conversion window, aggregation type, test account filter, attribution type options
image
3. Reconcile the same in the metric selector, so this selection happens only in one place
4. Disallow having global properties on experiments (which were used to create feature flags). 

This kind of funnel now works:

image

TODO: 

~~1. fix hogql aggregation update: it's not submitting / not changing the name of the aggregation label~~
~~3. Fix / write tests for hogql aggregation~~
4. Check with @thmsobrmlr whether the metric consolidation I've done is correct, and whether I can use a `querySource` in place of a `query` / `setQuery` pair. (P.s: Great work with building all the query compatible components, made adding these few new ones so seamless!)

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by neilkakkar** [comment]
@thmsobrmlr I still have some hogql testing to do (choosing a hogql expression to aggregate by doesn't really work), but was hoping to get an early review from you on the queryNodes usage, and if this makes sense to you

File: ee/clickhouse/views/test/__snapshots__/test_clickhouse_experiments.ambr [hunk 1]
```diff
@@ -412,9 +412,147 @@
...
+                                                                                     prop
+                                                                        ORDER BY timestamp DESC ROWS BETWEEN UNBOUNDED PRECEDING AND 0 PRECEDING) latest_1 ,
+                                                                       if(has([['test'], ['control'], ['']], prop), prop, ['Other']) as prop
+              FROM
+                (SELECT *,
+                        if(notEmpty(arrayFilter(x -> notEmpty(x), prop_vals)), prop_vals, ['']) as prop
+                 FROM
+                   (SELECT e.timestamp as timestamp,
+                           replaceRegexpAll(nullIf(nullIf(JSONExtractRaw(properties, '$account_id'), ''), 'null'), '^"|"$', '') as aggregation_target,
+                           pdi.person_id as person_id ,
```
Comment (neilkakkar) [inline ee/clickhouse/views/test/__snapshots__/test_clickhouse_experiments.ambr:523]: @mariusandra the aggregation by hogql seems to unnecessarily query & join the pdi table in funnels.

**Review by thmsobrmlr — APPROVED** [review]
> 4. Check with @thmsobrmlr whether the metric consolidation I've done is correct, and whether I can use a querySource in place of a query / setQuery pair. (P.s: Great work with building all the query compatible components, made adding these few new ones so seamless!)

Cool to see more options for the funnel here! I've had a look at that part of the code an all seems great to me.

With regard to the insights filters - the broader plan is to create some "official" query components that either take `insightProps` or a `query / setQuery` pair and can then be reused in other parts of the app. 

Background is that I gradually want to get away from importing components from random parts of the app. The components then would be marked as official e.g. by including them in a Storybook section and importing other components would be discouraged.

Just meant as an FYI and components used here would be good candidates for this.

**Comment by liyiy** [comment]
1. Nice work including the extra funnel options

2. I like that we're consolidating the person properties rollout to the feature flag! It'll give users a lot more flexibility if they don't want a 50/50 experiment for example, etc. 

It would be really nice though if you could just edit the flag directly in the experiment so you'd have context the entire time. That's probably a bit of work so in the meantime I'd recommend at least a simpler explanation on redirecting users to the feature flag, something like:
 
`Experiments use feature flags to select participants. For more advanced options like changing rollout percentages or targeting specific users, update the <name> feature flag for it.` (shorter is even better)

3. Below is an experiment I created and I edited the feature flag conditions for it. However the flag conditions displayed on the experiment shows `100% of users` which isn't super helpful/accurate and it'd also be nice to have the feature flag linked to it.

Screen Shot 2023-06-28 at 9 06 58 PM


4. To keep it consistent with the other Feature Success features, let's do a `Save as draft` instead of the initial `Save`
Screen Shot 2023-06-28 at 9 11 11 PM

**Review by liyiy — DISMISSED** [review]
1. Nice work including the extra funnel options

2. I like that we're consolidating the person properties rollout to the feature flag! It'll give users a lot more flexibility if they don't want a 50/50 experiment for example, etc. 

It would be really nice though if you could just edit the flag directly in the experiment so you'd have context the entire time. That's probably a bit of work so in the meantime I'd recommend at least a simpler explanation on redirecting users to the feature flag, something like:
 
`Experiments use feature flags to select participants. For more advanced options like changing rollout percentages or targeting specific users, update the <name> feature flag for it.` (shorter is even better)

3. Below is an experiment I created and I edited the feature flag conditions for it. However the flag conditions displayed on the experiment shows `100% of users` which isn't super helpful/accurate and it'd also be nice to have the feature flag linked to it.

Screen Shot 2023-06-28 at 9 06 58 PM


4. To keep it consistent with the other Feature Success features, let's do a `Save as draft` instead of the initial `Save`
Screen Shot 2023-06-28 at 9 11 11 PM

**Comment by neilkakkar** [comment]
For (2): the link is generated once you save the experiment, because the flag is created only on save.

Will update the text to make this clearer.

The others are issues that existed before this PR, so will fix in a follow up :)

**Comment by neilkakkar** [comment]
Actually, will fix participants in this PR too, since that confusion will exacerbate with this change

**Comment by neilkakkar** [comment]
It now accurately gets participants info from the flag:

image

and defers to flag when too many groups:

image




Also fixed the same on the feature flag list table

**Comment by neilkakkar** [comment]
Also added some backwards compatibility so existing experiments in draft mode aren't borked

**Review by liyiy — APPROVED** [review]

**Pull request merged**